src/test/java/com/epam/esm/dao/impl/CertificateDaoImplTest.java

```diff
@@ -1,21 +1,46 @@
 package com.epam.esm.dao.impl;
 
+import com.epam.esm.dao.DaoException;
+import com.epam.esm.model.Certificate;
+import org.junit.Before;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
+import org.mockito.Matchers;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.PreparedStatementCreator;
 
+import static com.epam.esm.dao.SqlProvider.REMOVE_CERTIFICATE;
 import static org.junit.jupiter.api.Assertions.*;
 
 class CertificateDaoImplTest {
+    private CertificateDaoImpl dao;
+    private JdbcTemplate jdbcTemplate;
+
+    @BeforeEach
+    void setup() {
+        dao = new CertificateDaoImpl();
```

Comment:
use autowired instead of new CertificateDaoImpl()

Reply:
Fixed